guacamole/src/main/java/org/apache/guacamole/rest/auth/TokenRESTService.java

```diff
@@ -122,6 +133,35 @@ private Credentials getCredentials(HttpServletRequest request,
 
     }
 
+    /**
+     * Alters letter casing of username, if configured.
+     * Configuration options:
+     * <ul>
+     *     <li>mixed-case (default)</li>
+     *     <li>lower-case</li>
+     *     <li>upper-case</li>
+     * </ul>
+     * @param credentials
+     *     The Credentials provided by the user in the most recent
+     *     authentication attempt.
+     * @throws GuacamoleException If an error occurs while parsing the value
+     *                            for the given property in
+     *                            guacamole.properties, or if the property is
+     *                            not specified.
+     */
+    private void adaptUsernameLetterCase(Credentials credentials) throws GuacamoleException {
+        switch (environment.getProperty(Environment.USERNAME_LETTER_CASE,DEFAULT_LETTER_CASE)){
+            case "lower-case":
+                Optional.ofNullable(credentials.getUsername()).map(String::toLowerCase).ifPresent(credentials::setUsername);
+                break;
+            case "upper-case":
+                Optional.ofNullable(credentials.getUsername()).filter(un -> !"guacadmin".equals(un)).map(String::toUpperCase).ifPresent(credentials::setUsername);
```

Comment:
We shouldn't expect any particular username - what is this check intending to accomplish?